From 73c007b045bfd79d55841c5e4908595a193329bb Mon Sep 17 00:00:00 2001
From: Sumit Jamadar <xkbr38@motorola.com>
Date: Mon, 28 Mar 2016 17:25:09 -0500
Subject: [PATCH 833/959] IKSWM-26923: msm: camera: Enable logsync in debug
 builds only

Enable logsync in debug builds only to avoid writable
sysfs file security issue.

Change-Id: I0801d26c7250e53f5508c1df98a702c82de90926
Signed-off-by: Sumit Jamadar <xkbr38@motorola.com>
Reviewed-on: http://gerrit.mot.com/837955
SLTApproved: Slta Waiver <sltawvr@motorola.com>
SME-Granted: SME Approvals Granted
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Igor Kovalenko <igork@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 drivers/media/platform/msm/camera_v2/Kconfig | 7 +++++++
 drivers/media/platform/msm/camera_v2/msm.c   | 6 ++++++
 2 files changed, 13 insertions(+)

diff --git a/drivers/media/platform/msm/camera_v2/Kconfig b/drivers/media/platform/msm/camera_v2/Kconfig
index f084d07eb6c..59ebee0ef52 100644
--- a/drivers/media/platform/msm/camera_v2/Kconfig
+++ b/drivers/media/platform/msm/camera_v2/Kconfig
@@ -235,6 +235,13 @@ config MSM_FD
 	 ---help---
 	    Enables support for the MSM FD face detection engine.
 
+config DEBUG_CAMERA
+	bool "Enable Camera Debug"
+	depends on MSMB_CAMERA && DEBUG_FS
+	default n
+	---help---
+		Enable camera debug support.
+
 config MSM_JPEGDMA
 	tristate "Qualcomm Technologies Inc. MSM Jpeg dma"
         depends on MSMB_CAMERA
diff --git a/drivers/media/platform/msm/camera_v2/msm.c b/drivers/media/platform/msm/camera_v2/msm.c
index adb2c180ec4..7b0041138af 100644
--- a/drivers/media/platform/msm/camera_v2/msm.c
+++ b/drivers/media/platform/msm/camera_v2/msm.c
@@ -1262,6 +1262,7 @@ static void msm_sd_notify(struct v4l2_subdev *sd,
 	}
 }
 
+#ifdef CONFIG_DEBUG_CAMERA
 static ssize_t write_logsync(struct file *file, const char __user *buf,
 		size_t count, loff_t *ppos)
 {
@@ -1285,11 +1286,14 @@ static ssize_t write_logsync(struct file *file, const char __user *buf,
 static const struct file_operations logsync_fops = {
 		.write = write_logsync,
 };
+#endif
 
 static int msm_probe(struct platform_device *pdev)
 {
 	struct msm_video_device *pvdev = NULL;
+#ifdef CONFIG_DEBUG_CAMERA
 	static struct dentry *cam_debugfs_root;
+#endif
 	int rc = 0;
 
 	msm_v4l2_dev = kzalloc(sizeof(*msm_v4l2_dev),
@@ -1373,6 +1377,7 @@ static int msm_probe(struct platform_device *pdev)
 	mutex_init(&v4l2_event_mtx);
 	INIT_LIST_HEAD(&ordered_sd_list);
 
+#ifdef CONFIG_DEBUG_CAMERA
 	cam_debugfs_root = debugfs_create_dir(MSM_CAM_LOGSYNC_FILE_BASEDIR,
 						NULL);
 	if (!cam_debugfs_root) {
@@ -1391,6 +1396,7 @@ static int msm_probe(struct platform_device *pdev)
 		pr_err("%s: failed to register ahb clocks\n", __func__);
 		goto v4l2_fail;
 	}
+#endif
 
 	goto probe_end;
 
-- 
2.11.0

